webapp/src/components/Metrics/PerformanceAnalysisTable.tsx

```diff
@@ -0,0 +1,658 @@
+  GridColumnsMenuItem,
+  GridColumnVisibilityModel,
+  GridRow,
+  GridSortCellParams,
+  GridSortModel,
+  gridStringOrNumberComparator,
+  HideGridColMenuItem,
+} from "@mui/x-data-grid";
+import DatasetSplitToggler from "components/Controls/DatasetSplitToggler";
+import DeltaComputationBar from "components/Metrics/DeltaComputationBar";
+import OutcomeIcon from "components/Icons/OutcomeIcon";
+import PipelineSelect from "components/PipelineSelect";
+import { Table, Column, RowProps } from "components/Table";
+import VisualBar from "components/VisualBar";
+import React from "react";
+import { Link } from "react-router-dom";
+import {
+  getConfigEndpoint,
+  getCustomMetricInfoEndpoint,
+  getMetricsPerFilterEndpoint,
+} from "services/api";
+import {
+  AvailableDatasetSplits,
+  DatasetSplitName,
+  MetricsPerFilterValue,
+} from "types/api";
+import { QueryPipelineState } from "types/models";
+import {
+  ALL_OUTCOMES,
+  ECE_TOOLTIP,
+  OUTCOME_COLOR,
+  OUTCOME_PRETTY_NAMES,
+  SMART_TAG_FAMILIES,
+  SMART_TAG_FAMILY_ICONS,
+  SMART_TAG_FAMILY_PRETTY_NAMES,
+} from "utils/const";
+import { formatRatioAsPercentageString } from "utils/format";
+import { constructSearchString } from "utils/helpers";
+
+const ROW_HEIGHT = 35;
+const OVERALL_ROW_ID = -1; // -1 so that the other rows can range from 0 - n-1
+const LABEL_WIDTH = 200;
+const LABEL_HEIGHT = 32;
+
+const pipelines = ["basePipeline", "comparedPipeline"] as const;
+const BASIC_FILTER_OPTIONS = ["label", "prediction"] as const;
+const OPTION_PRETTY_NAME = {
+  label: "Label",
+  prediction: "Prediction",
+  ...SMART_TAG_FAMILY_PRETTY_NAMES,
+} as const;
+
+const ColumnMenu = ({ hideMenu, currentColumn, open }: GridColumnMenuProps) => (
+  <GridColumnMenuContainer
+    hideMenu={hideMenu}
+    currentColumn={currentColumn}
+    open={open}
+  >
+    <HideGridColMenuItem onClick={hideMenu} column={currentColumn} />
+    <GridColumnsMenuItem onClick={hideMenu} column={currentColumn} />
+  </GridColumnMenuContainer>
+);
+
+type FilterByViewOption =
+  | typeof BASIC_FILTER_OPTIONS[number]
+  | typeof SMART_TAG_FAMILIES[number];
+
+type Props = {
+  jobId: string;
+  pipeline: Required<QueryPipelineState>;
+  availableDatasetSplits: AvailableDatasetSplits | undefined;
+  isLoading: boolean;
+};
+
+type Row = {
+  id: number;
+  basePipeline: MetricsPerFilterValue;
+  comparedPipeline?: MetricsPerFilterValue;
+};
+
+const PerformanceAnalysisTable: React.FC<Props> = ({
+  jobId,
+  pipeline,
+  availableDatasetSplits,
+  isLoading,
+}) => {
+  const [selectedDatasetSplit, setSelectedDatasetSplit] =
+    React.useState<DatasetSplitName>("eval");
+
+  const [selectedMetricPerFilterOption, setSelectedMetricPerFilterOption] =
+    React.useState<FilterByViewOption>("label");
+
+  const [comparedPipeline, setComparedPipeline] = React.useState<
+    number | undefined
+  >();
+
+  const { data: config } = getConfigEndpoint.useQuery(
+    { jobId },
+    { skip: jobId === undefined }
+  );
```

Comment:
The type of `jobId` is `string`, so it (supposedly) can't be `undefined` and this `skip` seems irrelevant. Am I missing a special reason for including it? If so, a comment should explain it.
```suggestion
  const { data: config } = getConfigEndpoint.useQuery({ jobId });
```

Reply:
> But if the URL seems to be this way (maybe the user edits by themselves like this http://localhost:3005/undefined/performance_analysis?pipeline_index=0 ), the page gets blocked.

In that case, `jobId` is `"undefined"`, not `undefined`, so it still wouldn't skip. Its type is `jobId: string;`, so it's not possible that `jobId === undefined`.